collector/internal/telemetryapi/listener.go

```diff
@@ -114,13 +126,25 @@ func (s *Listener) Shutdown() {
 	}
 }
 
-func (s *Listener) Wait(ctx context.Context, reqID string) error {
+func (s *Listener) handleEvent(ctx context.Context, e Event) error {
+	handlerLock.Lock()
```

Comment:
A read lock is probably sufficient here, but it would only be useful if multiple concurrent reads are possible.

Reply:
This should be the only thing reading from the handler list and ideally nothing is writing to it after initialization, so there should be very little to no contention on this lock.

Reply:
Agreed that lock contention should be minimal, changed the code to use a read lock nonetheless